core-tests/src/test/java/org/mule/runtime/core/processor/interceptor/ReactiveInterceptorAdapterTestCase.java

```diff
@@ -340,33 +364,62 @@ public void interceptorMutatesEventAroundAfterFailWithCause() throws Exception {
       @Override
       public CompletableFuture<InterceptionEvent> around(Map<String, Object> parameters, InterceptionEvent event,
                                                          InterceptionAction action) {
-        action.fail(cause);
-        return supplyAsync(() -> {
-          event.message(Message.of(TEST_PAYLOAD));
-          return event;
-        });
+        event.message(Message.of(TEST_PAYLOAD));
+        return action.fail(cause);
       }
     });
     startFlowWithInterceptors(interceptor);
 
-    Event result = process(flow, eventBuilder().message(Message.of("")).build());
-    assertThat(result.getMessage().getPayload().getValue(), is(TEST_PAYLOAD));
-    assertThat(result.getError().isPresent(), is(true));
-    assertThat(result.getError().get().getCause(), is(instanceOf(RuntimeException.class)));
-    assertThat(result.getError().get().getErrorType().getNamespace(), is(equalTo("MULE")));
-    assertThat(result.getError().get().getErrorType().getIdentifier(), is(equalTo("UNKNOWN")));
+    expected.expectCause(sameInstance(cause));
+    try {
+      process(flow, eventBuilder().message(Message.of("")).build());
+    } finally {
+      InOrder inOrder = inOrder(processor, interceptor);
 
-    InOrder inOrder = inOrder(processor, interceptor);
+      inOrder.verify(interceptor).before(mapArgWithEntry("param", ""), any());
+      inOrder.verify(interceptor).around(mapArgWithEntry("param", ""), any(), any());
+      inOrder.verify(processor, never()).process(any());
+      inOrder.verify(interceptor).after(argThat(interceptionHasPayloadValue(TEST_PAYLOAD)), eq(of(cause)));
+    }
+  }
 
-    inOrder.verify(interceptor).before(mapArgWithEntry("param", ""), any());
-    inOrder.verify(interceptor).around(mapArgWithEntry("param", ""), any(), any());
-    inOrder.verify(processor, never()).process(any());
-    inOrder.verify(interceptor).after(argThat(interceptionHasPayloadValue(TEST_PAYLOAD)), eq(empty()));
+  @Test
+  public void interceptorMutatesEventAroundAfterFailWithCauseNoMock() throws Exception {
+    Throwable cause = new RuntimeException("");
+    ProcessorInterceptor interceptor = new ProcessorInterceptor() {
+
+      @Override
+      public CompletableFuture<InterceptionEvent> around(Map<String, Object> parameters, InterceptionEvent event,
+                                                         InterceptionAction action) {
+        event.message(Message.of(TEST_PAYLOAD));
+        return action.fail(cause);
+      }
+    };
+    startFlowWithInterceptors(interceptor);
+
+    expected.expectCause(sameInstance(cause));
+    process(flow, eventBuilder().message(Message.of("")).build());
```

Comment:
I'm guessing this throws a MessagingException in which cause shouldn't you also assert about the error type of the associated event?